src/runtime_src/core/tools/xbutil2/OO_MemWrite.cpp

```diff
@@ -142,13 +180,26 @@ XBU::verbose("SubCommand option: read mem");
   XBU::verbose(boost::str(boost::format("Device: %s") % xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(device))));
   XBU::verbose(boost::str(boost::format("Address: %s") % addr));
   XBU::verbose(boost::str(boost::format("Size: %s") % size));
+  XBU::verbose(boost::str(boost::format("Block count: %s") % count));
   XBU::verbose(boost::str(boost::format("Fill pattern: %s") % m_fill));
+  XBU::verbose(boost::str(boost::format("Input File: %s") % m_inputFile));
 
-  //read mem
+  //write mem
   XBU::xclbin_lock xclbin_lock(device);
   
   try {
-    xrt_core::mem_write(device.get(), addr, size, fill_byte);
+    for(int c = 0; c < count; c++) {
+      if(!m_fill.empty())
+        xrt_core::mem_write(device.get(), addr, size, fill_byte);
+      else {
+        std::vector<char> buffer(size);
+        auto input_size = input_stream->read(buffer.data(), size).gcount();
+        xrt_core::mem_write(device.get(), addr, size, buffer);
+        if (input_size != size)
+          break; // partial read and break the loop
+      }
+      addr +=size;
```

Comment:
It might be nice to have a status update on what is being written (similar to what was suggested earlier for the read operation).